src/service-worker.ts

```diff
@@ -98,97 +86,76 @@ export async function generateServiceWorkerConfig(
+  let staticFileGlobs = Array.from(workboxConfig.globPatterns || []);
   const precachedAssets = (options.bundled) ?
-      getBundledPrecachedAssets(project) :
-      getPrecachedAssets(depsIndex, project);
+    getBundledPrecachedAssets(project) :
+    getPrecachedAssets(depsIndex, project);
 
   staticFileGlobs = staticFileGlobs.concat(precachedAssets);
   staticFileGlobs = staticFileGlobs.map((filePath: string) => {
     if (filePath.startsWith(project.config.root)) {
       filePath = filePath.substring(project.config.root.length);
     }
-    return path.join(buildRoot, filePath);
+    return removeLeadingSlash(filePath);
   });
 
-  if (swPrecacheConfig.navigateFallback === undefined) {
+  if (workboxConfig.globDirectory === undefined) {
+    workboxConfig.globDirectory = buildRoot;
+  }
+
+  if (workboxConfig.importScripts === undefined) {
+    // Map all application routes to the entrypoint.
+    workboxConfig.importScripts =
+      ['https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-sw.js'];
+  }
+
+  if (workboxConfig.navigateFallback === undefined) {
     // Map all application routes to the entrypoint.
-    swPrecacheConfig.navigateFallback =
-        path.relative(project.config.root, project.config.entrypoint);
+    workboxConfig.navigateFallback =
+      path.relative(project.config.root, project.config.entrypoint);
   }
 
-  if (swPrecacheConfig.navigateFallbackWhitelist === undefined) {
+  if (workboxConfig.navigateFallbackWhitelist === undefined) {
     // Don't fall back to the entrypoint if the URL looks like a static file.
     // We want those to 404 instead, since they are probably missing assets,
     // not application routes. Note it's important that this matches the
     // behavior of prpl-server.
-    swPrecacheConfig.navigateFallbackWhitelist = [hasNoFileExtension];
+    workboxConfig.navigateFallbackWhitelist = [hasNoFileExtension];
   }
 
-  if (swPrecacheConfig.directoryIndex === undefined) {
-    // By default, sw-precache maps any path ending with "/" to "index.html".
-    // This is a reasonable default for matching application routes, but 1) our
-    // entrypoint might not be called "index.html", and 2) this case is already
-    // handled by the navigateFallback configuration above. Simplest to just
-    // disable this feature.
-    swPrecacheConfig.directoryIndex = '';
-  }
-
-  // swPrecache will determine the right urls by stripping buildRoot.
-  // NOTE:(usergenic) sw-precache generate() apparently replaces the
-  // prefix on an already posixified version of the path on win32.
-  //
-  // We include a trailing slash in `stripPrefix` so that we remove leading
-  // slashes on the pre-cache asset URLs, hence producing relative URLs
-  // instead of absolute. We want relative URLs for builds mounted at non-root
-  // paths. Note that service worker fetches are relative to its own URL.
-  swPrecacheConfig.stripPrefix = addTrailingSlash(posixifyPath(buildRoot));
```

Comment:
this is something i am missing on right now, and would probably add as another manifestTransform